lib/spack/spack/build_systems/intel.py

```diff
@@ -1241,6 +1247,33 @@ def configure_rpath(self):
             with open(compiler_cfg, 'w') as fh:
                 fh.write('-Xlinker -rpath={0}\n'.format(compilers_lib_dir))
 
+    # Need a copy of the auto_dispatch_options to be used in the subclasses
+    auto_dispatch_options = auto_dispatch_options
+
+    @run_after('install')
+    def configure_auto_dispatch(self):
+        if self._has_compilers:
+            if ('auto_dispatch=none' in self.spec):
+                return
+
+            compilers_bin_dir = self.component_bin_dir('compiler')
+
+            for compiler_name in 'icc icpc ifort'.split():
+                f = os.path.join(compilers_bin_dir, compiler_name)
+                if not os.path.isfile(f):
```

Comment:
There may be intel installations which are provided by modules and cannot be located, if you can place the configuration files in the build directory, I think you don't have to worry about whether the executables can be found (that will be dealt with elsewhere).

Reply:
If an Intel installation is provided by modules then this code would not be in play because the compiler would already be installed. Again, I may be misunderstanding.

Reply:
Yes: sorry.